docs/guidelines.md

```diff
@@ -0,0 +1,87 @@
+- `approve_with_comment`: **COULD** be a best solution, **MUST** be with comment
+- `disapprove_with_comment`: **MUST** be with comment
+- `refer_to_mentor`: **COULD** be with comment
+
+## Conditions and outcome
+
+For feedback generated by automated mentoring:
+
+- it **SHOULD** `disapprove_with_comment` when someone isn't using one of the
+  subjects the exercise is supposed to teach,
+- each `disapprove_with_comment` **MUST** has at least one comment,
+- each `disapprove_with_comment` **SHOULD** steer a student towards an `approve_`,
+- a student **SHOULD NOT** get a `refer_to_mentor` in a newer iteration, if they've
+  had a `disapprove_with_comment` in a previous iteration,
+- a `refer_to_mentor` **COULD** have one or more comments which will be given to
+  the mentor.
+
+There is no minimum or maximum amount of :speech_balloon: comments. However, each
+comment **SHOULD** be aiding the student towards an approvable solution.
+
+## Approvability
+
+Given the above, and to re-iterate that we focus on language fluency:
+
+### Incorrect Naming or Casing
+
+In general, if it's a _language feature_ that will be caught by a compiler or parser
+or based on official rules (which means there is a dependency on correctness in
+tools), you should `disapprove_with_comment`, preferably linking to the official
+rules. If it's a _stylistic_ choice, or there are _competing standards_, do not
+remark at all.
+
+> - :-1: dissapprove if there are official *strict* guidelines on naming conventions
+> - :+1: approve if there are only _defacto_ standard(s) and everything else is fine.
+> - :speech_balloon: leave a comment if there is something noteworthy
+> - :no_bell: ignore stylistic issues if it's not integral to the language
+
+#### Examples
+
+- **Ruby** has a language feature where `_` is treated differently,
+  - :-1: If a student uses `_` for a variable name, but then uses it.
+- **Ruby** recognises `constants` only if they start with a **C**apital Letter,
+  - :-1: If a student uses `snake_case` for a `class` name
+- **Ruby** has _defacto_ standards on `cAsInG` and `name-ing`,
+  - :no_bell: If a student uses `camelCase` for a variable, leave it be.
+  - :speech_balloon: You may guide students that `snake_case` is to be expected by
+    most IDEs and highlighting on exercism in code blocks for variables.
+- **JavaScript** IDEs highlight variables which are not used, except for those
+  prefixed with an underscore (`_`).
+  - :speech_balloon: Note that this behaviour exist so it might help them to use a
+    different naming strategy. They might think that prefixing with `_` means 
+    `private`, which is not the case in JavaScript.
+- **TypeScript** has a _defacto_ standard lint tool provided by Palantir,
+  - :no_bell: If a student does not follow these rules as the lint tool is not 
+    official. In fact, there are multiple linters out there, with mutually exclusive
+    rules.
+- **Go** has very strict rules around naming and other linting.
```

Comment:
```suggestion
- **Go** has very strict rules around formatting and linting.
```
Or change this, then we can keep `gofmt` below.

Reply:
Nah, I liked the other one 👍 . We can of course add both 📦